### ISSUE TITLE
NVDA repeats previous line when caret moves to last blank line in JetBrains Rider (Java Access Bridge)

### ISSUE THREAD

**Issue by Jonikster** [nvaccess/nvda#19516]
### Brief summary

If the last line of the source file is empty, NVDA doesn't read blank as expected, but the previous line instead.

### Steps to reproduce

1. Open any source file in JetBrains Rider 2025.3.1;
2. Make 3 lines in editor, 1 and 3 are empty and something on the second line;
3. Being on the second line, press down arrow.

### Actual behavior

Says the content of the second line.

### Expected behavior

Says blank as line is empty.

### NVDA logs, crash dumps and other attachments

INFO - __main__ (05:46:45.698) - MainThread (5380):
Starting NVDA version 2025.3.2 x86
INFO - core.main (05:46:45.723) - MainThread (5380):
Config dir: C:\Users\John\AppData\Roaming\nvda
INFO - config.ConfigManager._loadConfig (05:46:45.727) - MainThread (5380):
Loading config: C:\Users\John\AppData\Roaming\nvda\nvda.ini
INFO - core.main (05:46:46.365) - MainThread (5380):
Windows version: Windows 10 21H2 (10.0.19044.6332) workstation AMD64
INFO - core.main (05:46:46.365) - MainThread (5380):
Using Python version 3.11.9 (tags/v3.11.9:de54cf5, Apr  2 2024, 10:00:00) [MSC v.1938 32 bit (Intel)]
INFO - core.main (05:46:46.365) - MainThread (5380):
Using comtypes version 1.4.11
INFO - core.main (05:46:46.367) - MainThread (5380):
Using configobj version 5.1.0 with validate version 1.0.1
INFO - synthDrivers.espeak.SynthDriver.__init__ (05:46:46.597) - MainThread (5380):
Using eSpeak NG version 1.52.0
INFO - synthDriverHandler.setSynth (05:46:46.662) - MainThread (5380):
Loaded synthDriver espeak
WARNING - mathPres.initialize (05:46:46.669) - MainThread (5380):
MathPlayer 4 not available
INFO - core._setUpWxApp (05:46:46.669) - MainThread (5380):
Using wx version 4.2.2 msw (phoenix) wxWidgets 3.2.6 with six version 1.17.0
INFO - brailleInput.initialize (05:46:46.671) - MainThread (5380):
Braille input initialized
INFO - braille.initialize (05:46:46.671) - MainThread (5380):
Using liblouis version 3.34.0
INFO - braille.initialize (05:46:46.671) - MainThread (5380):
Using pySerial version 3.5
INFO - braille.BrailleHandler._setDisplay (05:46:46.673) - MainThread (5380):
Loaded braille display driver 'noBraille', current display has 0 cells.
INFO - core.main (05:46:46.825) - MainThread (5380):
Java Access Bridge support initialized
INFO - UIAHandler.UIAHandler.MTAThreadFunc (05:46:46.888) - UIAHandler.UIAHandler.MTAThread (15196):
UIAutomation: IUIAutomation6
INFO - _remoteClient.client.RemoteClient.__init__ (05:46:47.681) - MainThread (5380):
Initializing NVDA Remote client
INFO - core.main (05:46:47.682) - MainThread (5380):
NVDA initialized
INFO - globalCommands.script_navigatorObject_devInfo (05:46:58.023) - MainThread (5380):
Developer info for navigator object:
name: 'Editor for Program.cs'
role: EDITABLETEXT
processID: 8396
roleText: None
states: FOCUSABLE, EDITABLE, FOCUSED, MULTILINE
isFocusable: True
hasFocus: True
Python object: <NVDAObjects.Dynamic_EditableTextWithoutAutoSelectDetectionJAB object at 0x07524770>
Python class mro: (<class 'NVDAObjects.Dynamic_EditableTextWithoutAutoSelectDetectionJAB'>, <class 'NVDAObjects.behaviors.EditableTextWithoutAutoSelectDetection'>, <class 'editableText.EditableTextWithoutAutoSelectDetection'>, <class 'NVDAObjects.behaviors.EditableText'>, <class 'NVDAObjects.behaviors.EditableTextWithSuggestions'>, <class 'NVDAObjects.behaviors.InputFieldWithSuggestions'>, <class 'NVDAObjects.behaviors.EditableTextBase'>, <class 'editableText.EditableText'>, <class 'NVDAObjects.JAB.JAB'>, <class 'NVDAObjects.window.Window'>, <class 'NVDAObjects.NVDAObject'>, <class 'documentBase.TextContainerObject'>, <class 'baseObject.ScriptableObject'>, <class 'baseObject.AutoPropertyObject'>, <class 'garbageHandler.TrackedObject'>, <class 'object'>)
description: ''
location: RectLTWH(left=657, top=180, width=1119, height=451)
value: None
TextInfo: <class 'NVDAObjects.JAB.JABTextInfo'>
appModule: AppModule(appModuleHandler, appName='rider64', processID=8396)
appModule.productName: 'JetBrains Rider'
appModule.productVersion: '253.29346.144.0-RD'
appModule.helperLocalBindingHandle: None
appModule.appArchitecture: 'AMD64'
windowHandle: 395006
windowClassName: 'SunAwtFrame'
windowControlID: 0
windowStyle: 382664704
extendedWindowStyle: 256
windowThreadID: 3856
windowText: 'ConsoleApp1 – Program.cs'
displayText: ''

### NVDA type

installed copy

### NVDA version

2025.3.2

### Have you tried any other versions of NVDA? If so, please report their behaviors.

I used 2025.3, then updated to 2025.3.2 to change if fixed

### Windows version

Windows 10 IoT LTSC 21H2 Build 19044.6332

### Name and version of other software in use when reproducing the issue

JetBrains Rider 2025.3.1

### Does the issue still occur after restarting your computer?

I have restarted my computer and the issue still occurs

### If NVDA add-ons are disabled, is your problem still occurring?

I have restarted NVDA with add-ons disabled and the issue still occurs

### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?

I have run the COM Registration Fixing Tool and the issue still occurs

**Comment by Jonikster** [comment]
I have no idea if it's NVDA or Rider issue as it's Java Access Bridge issue I beleive.

**Comment by seanbudd** [comment]
Is this a duplicate of #17027?

**Comment by seanbudd** [comment]
Can you please test the latest [alpha snapshot](download.nvaccess.org/snapshots/alpha/) and let us know if the behaviour is still there?

**Comment by dmitrii-drobotov** [comment]
There are two separate issues related to last lines with JAB text areas: #9376 and #17027. This one looks more like a duplicate of #9376. I tried to fix it back then but it made the other issue #17027 more apparent, so my fix was reverted. I couldn't find a solution for both cases at the time and didn't investigate further. Both of them should reproduce on the latest NVDA version.
